### PR TITLE
Code for axon mapping paper

### PR DESCRIPTION
**Pull request by tathey1**
Multiple items addressed:

- Archived `adaptive_thresh` and moved archive directory into root directory
- Extended `fit_spline` so it can use either b-splines OR cubic Hermite splines. Cubic Hermite splines are used when nodes have associated derivative estimates (and thus can be fit with hermite splines)
- New map_neurons module defines a `DiffeomorphismAction` interface which can be used to map positions/derivatives of curves in 3d space. This transformation is done via `transform_GeometricGraph`.
- I implement the `DiffeomorphismAction` interface in `CloudReg_Transform` which is able to read the registrations computed by [CloudReg](https://github.com/neurodata/CloudReg). Briefly, CloudReg estimates an affine and a non-affine (aka _velocity based_) transformation. The velocity based component is represented by a series of small vector field encoded deformations.
- The `map_neurons` module also can estimate derivatives from a sequence of positions using the nonuniform finite-difference method from Svndquist & Veronis 1970.
- `NeuronTrace` now only rounds node coordinates through an argument `round`, rather than doing so automatically.
- Notebooks using the mapping code in experiments. Of note is `map_neurons.ipynb` which walk through figures in my manuscript. Other notebooks are more scratch, for future reference.
- Tests for the above mentioned code.

### CODE REVIEW

**Comment by tathey1** [comment]
@JacopoTeneggi review whatever you have time/energy for, no pressure. Even if you only verify that the files/code changed roughly matches the bullet points in my PR description, that would be much appreciated. Note: codecov went down because I removed some files that had above average coverage.

File: brainlit/map_neurons/map_neurons.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+            )
+        norms = np.linalg.norm(diffs, axis=1)
+        derivs = np.divide(diffs, np.array([norms]).T)
+    else:
+        raise ValueError(f"Invalid deriv_method argument: {deriv_method}")
+
+    return derivs
+
+
+def transform_GeometricGraph(
```
Comment (JacopoTeneggi) [inline brainlit/map_neurons/map_neurons.py]: lowercase name of function

```suggestion
def transform_geometricgraph(
```

File: brainlit/map_neurons/map_neurons.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+
+def compute_derivs(
+    us: np.array,
+    tck: tuple = None,
+    positions: np.array = None,
+    deriv_method: str = "difference",
+) -> np.array:
+    """Estimate derivatives of a sequence of points. Derivatives can be estimated in two ways:
+    - For curves parameterized by scipy's spline API, spline estimation uses scipy's derivative computation
+    - For a sequence of points, the finite-difference method from Svndquist & Veronis 1970 for nonuniform intervals is used.
```
Comment (JacopoTeneggi) [inline brainlit/map_neurons/map_neurons.py]: If possible, include ref or title of paper

File: docs/reference/map_neurons.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
+Mapping Neurons
```
Comment (JacopoTeneggi) [inline docs/reference/map_neurons.rst:1]: - Add `similaritymeasures` as a dependency
- Add `statannotations` / `statannot` as a dependency
- Notebook fails on `Figure 4b,c,d) CloudReg + Real Neuron with error ```Unable to open file ... brainlit/experiments/register_axons/data/mapping-files/downloop_1_v.mat'```

**Review by JacopoTeneggi — CHANGES_REQUESTED** [review]

File: brainlit/map_neurons/map_neurons.py (hunk 1 above)
Comment (tathey1) [reply brainlit/map_neurons/map_neurons.py]: changed

File: brainlit/map_neurons/map_neurons.py (hunk 2 above)
Comment (tathey1) [reply brainlit/map_neurons/map_neurons.py]: APA ref added (not sure if I need to add italics or something?)

File: docs/reference/map_neurons.rst (hunk 3 above)
Comment (tathey1) [reply docs/reference/map_neurons.rst:1]: added dependencies to `setup.py`. Also added data file with git lfs and documentation to unzip the file before running notebook.

**Review by JacopoTeneggi — APPROVED** [review]

**Pull request merged**